### PR TITLE
Better output for Pkg.status()

### PR DESCRIPTION
**Pull request by jakebolewski**
Show a more descriptive output for status() that highlights problematic packages.

### CODE REVIEW

File: base/pkg/entry.jl [hunk 1]
```diff
@@ -174,13 +174,13 @@ function status(io::IO, pkg::AbstractString, ver::VersionNumber, fix::Bool)
             isfile("METADATA",pkg,"url") || push!(attrs,"unregistered")
             LibGit2.isdirty(prepo) && push!(attrs,"dirty")
             isempty(attrs) || print(io, " (",join(attrs,", "),")")
-        catch
-            print(io, "broken-repo (unregistered)")
+        catch err
```
Comment (tkelman) [inline base/pkg/entry.jl]: should this show what the error was?

Comment (jakebolewski) [inline base/pkg/entry.jl]: it could, but this is really just a sign that the git-repo is corrupted

Comment (jakebolewski) [inline base/pkg/entry.jl]: Also, this is just a summary table of the currently installed packages, so throwing an error half way through seems weird.  The assumption that was unchecked here was that is was a git repo at all which should be fixed now.

Comment (tkelman) [inline base/pkg/entry.jl]: makes sense, I prefer this over https://github.com/JuliaLang/julia/issues/13639 so +1

**Comment by tkelman** [comment]
why did you get rid of the important part of this?

**Pull request merged**